internal/pkg/cli/svc_deploy.go

```diff
@@ -570,30 +581,34 @@ func (o *deploySvcOpts) deploySvc(addonsURL string) error {
 		var errEmptyCS *awscloudformation.ErrChangeSetEmpty
 		if errors.As(err, &errEmptyCS) {
 			if o.forceNewUpdate {
-				// Force update ECS service if --force is set and change set is empty.
-				o.spinner.Start(fmt.Sprintf(fmtForceUpdateSvcStart, color.HighlightUserInput(o.name), color.HighlightUserInput(o.envName)))
-				if err = o.svcUpdater.ForceUpdateService(o.appName, o.envName, o.name); err != nil {
-					errLog := fmt.Sprintf(fmtForceUpdateSvcFailed, color.HighlightUserInput(o.name),
-						color.HighlightUserInput(o.envName), err)
-					var errWaitUpdateTimeout *awsecs.ErrWaitServiceStableTimeout
-					if errors.As(err, &errWaitUpdateTimeout) {
-						errLog = fmt.Sprintf("%s  Run %s to check for the fail reason.\n", errLog,
-							color.HighlightCode(fmt.Sprintf("copilot svc status --name %s --env %s", o.name, o.envName)))
-					}
-					o.spinner.Stop(log.Serror(errLog))
-					return fmt.Errorf("force an update for service %s: %w", o.name, err)
-				}
-				o.spinner.Stop(log.Ssuccessf(fmtForceUpdateSvcComplete, color.HighlightUserInput(o.name), color.HighlightUserInput(o.envName)))
-				return nil
-			} else {
-				log.Warningf("Set --%s to force an update for the ECS service.\n", forceFlag)
+				return o.forceDeploy()
 			}
+			log.Warningf("Set --%s to force an update for the service.\n", forceFlag)
 		}
 		return fmt.Errorf("deploy service: %w", err)
 	}
 	return nil
 }
 
+func (o *deploySvcOpts) forceDeploy() error {
+	// Force update the service if --force is set and change set is empty.
+	o.spinner.Start(fmt.Sprintf(fmtForceUpdateSvcStart, color.HighlightUserInput(o.name), color.HighlightUserInput(o.envName)))
+	if err := o.svcUpdater.ForceUpdateService(o.appName, o.envName, o.name); err != nil {
+		errLog := fmt.Sprintf(fmtForceUpdateSvcFailed, color.HighlightUserInput(o.name),
+			color.HighlightUserInput(o.envName), err)
+		var errWaitUpdateTimeout *awsecs.ErrWaitServiceStableTimeout
+		var errWaitOperationFailed *awsapprunner.ErrWaitServiceOperationFailed
+		if errors.As(err, &errWaitUpdateTimeout) || errors.As(err, &errWaitOperationFailed) {
```

Comment:
Check out this refactor boss: https://play.golang.org/p/VJzxx9PDXG8

What we do here is 1) each error implements the `timeoutError` interface, 2) we check if the error implements the interface in `errors.As`:
```go
var terr timeoutError
if errors.As(err, &terr) {
    ..
}
```

I like this advice from the [Practical Object-Oriented Design book](https://www.poodr.com/) for when we see ourselves checking multiple types:
> When a sending object checks the class of a receiving object to determine what message to send, you have overlooked a duck type.

For Go this means, if we see any statements that have to check multiple data type there is probably an interface that they can implement

Reply:
lol

Reply:
Oooh i see what you mean. Good idea!